package.json

```diff
@@ -8,10 +8,15 @@
     "start": "next start"
   },
   "dependencies": {
-    "next": "10.0.6",
+    "next": "^10.0.6",
     "next-images": "^1.7.0",
-    "react": "17.0.1",
-    "react-dom": "17.0.1",
+    "react": "^17.0.1",
+    "react-dom": "^17.0.1",
     "styled-components": "^5.2.1"
+  },
+  "devDependencies": {
+    "@types/node": "^14.14.27",
+    "@types/react": "^17.0.2",
+    "typescript": "^4.1.5"
```

Comment:
Adicionei o typescript só pra mostrar que não muda nada hahaha, tu só ganha super poderes e teu código continua igual... só troquei o nome dos arquivos de .js para .tsx... 

O NextJS por padrão roda com os dois... se teu arquivo do pages/index for um .tsx o próprio next já te avisa pra instalar `yarn install -D typescript @types/node @types/react`

Pq ele faz isso? Pois assim ele identifica todos os pormenores do node e do react pra te dar sugestões automáticas de código para tu ter menos erros. Ele te ajuda, quase te força, a ter um código com melhor qualidade... pois sempre que tu faz algo "que não deveria em javascript" o typescript te barra e diz que isso é um erro